tests/test_graphql_api/test_application_event_schedule/test_approve.py

```diff
@@ -184,3 +205,331 @@ def test_cannot_approve_if_application_is_not_in_unallocated(graphql):
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The response complains about the allocated time being invalid.
+    assert response.field_error_messages() == [
+        "Cannot allocate schedule for this day and time period. "
+        "Given time period does not fit within applicants wished periods.",
+    ]
+
+
+def test_approve_schedule__can_approve_event_outside_of_wished_period_with_force_flag(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation()
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    # when:
+    # - The user tries to allocate the schedule, but it does not fall
+    #   within the applicants the wished periods, but uses the force flag.
+    input_data = approve_data(application, begin="14:00:00", end="15:00:00", force=True)
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The allocation is successful
+    assert response.has_errors is False, response
+
+
+def test_approve_schedule__approved_time_falls_on_two_back_to_back_wished_times(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation()
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    ApplicationEventScheduleFactory.create(
+        application_event=application.application_events.first(),
+        day=WeekdayChoice.MONDAY,
+        begin=datetime.time(14, 0, tzinfo=get_default_timezone()),
+        end=datetime.time(16, 0, tzinfo=get_default_timezone()),
+    )
+
+    # when:
+    # - The user tries to allocate the schedule, and it falls
+    #   on two of the applicants the wished periods.
+    input_data = approve_data(application, begin="13:00:00", end="15:00:00")
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The allocation is successful
+    assert response.has_errors is False, response
+
+
+def test_approve_schedule__approved_time_falls_on_two_separated_wished_times(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation()
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    ApplicationEventScheduleFactory.create(
+        application_event=application.application_events.first(),
+        day=WeekdayChoice.MONDAY,
+        begin=datetime.time(14, 30, tzinfo=get_default_timezone()),
+        end=datetime.time(16, 0, tzinfo=get_default_timezone()),
+    )
+
+    # when:
+    # - The user tries to allocate the schedule, and it falls
+    #   on two of the applicants the wished periods.
+    input_data = approve_data(application, begin="13:00:00", end="15:00:00")
```

Comment:
Could test with the force flag to prove that it does not allow overriding this check.

Reply:
Done